pkg/apiserver/resthandler_test.go

```diff
@@ -324,6 +324,54 @@ func TestPatchResourceWithVersionConflict(t *testing.T) {
 	tc.Run(t)
 }
 
+func TestNoAPIVersionMixup(t *testing.T) {
+	namespace := "bar"
+	name := "foo"
+	uid := types.UID("uid")
+
+	tc := &patchTestCase{
+		name: "TestPatchResourceWithVersionConflict",
+
+		startingPod: &v1.Pod{},
+		changedPod:  &v1.Pod{},
+		updatePod:   &v1.Pod{},
+
+		expectedPod: &api.Pod{},
+	}
+
+	// DeprecatedServiceAccount is only present in v1.PodSpec, but not in api.PodSpec.
+	// If patchResource unmarshals patched JSON to internal, this patch will be ignored.
+
+	tc.startingPod.Name = name
+	tc.startingPod.Namespace = namespace
+	tc.startingPod.UID = uid
+	tc.startingPod.ResourceVersion = "1"
+	tc.startingPod.APIVersion = "v1"
+	tc.startingPod.Spec.DeprecatedServiceAccount = "acct"
+
+	tc.changedPod.Name = name
+	tc.changedPod.Namespace = namespace
+	tc.changedPod.UID = uid
+	tc.changedPod.ResourceVersion = "1"
+	tc.changedPod.APIVersion = "v1"
+	tc.changedPod.Spec.DeprecatedServiceAccount = "acct"
```

Comment:
I'm not seeing any difference between startingPod and changedPod... won't that make a no-op patch?